containers/ftp/ftp_patch.rb

```diff
@@ -0,0 +1,62 @@
+require 'ipaddr'
+require 'net/ftp'
+
+# N.B. look at Socket AddrInfo stdlib, might be a better way to do this
+class IPAddr
+  IP4_PRIVATE_RANGES = [
+    IPAddr.new('10.0.0.0/8'),
+    IPAddr.new('172.16.0.0/12'),
+    IPAddr.new('192.168.0.0/16'),
+    IPAddr.new('127.0.0.0/8')
+  ]
+
+  IP6_PRIVATE_RANGES = [IPAddr.new('fc00::/7'), IPAddr.new('::1')]
+
+  def private?
+    ranges = self.ipv6? ? IP6_PRIVATE_RANGES : IP4_PRIVATE_RANGES
+    ranges.each { |ipr| return true if ipr.include?(self) }
+    return false
+  end
+
+  def public?
+    !private?
+  end
+end
+
+module Net
+  class FTP # provide a masquerade_host to be able to override incorrect PORT response
```

Comment:
This is a reference to how FTP servers like proftpd are configured, where in the server's config it can be set up wrong.
That is why the value is not present here, it is explaining we need this to work around badly configures servers (it happens).

Reply:
Updated comments to explain the overrides in Net::FTP better